docs/Node-for-Formula-Authors.md

```diff
@@ -102,7 +102,19 @@ class Foo < Formula
 
   def install
     system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+
+    # "link" the executable `foo`, ensuring that the version of node installed by Homebrew is used
+    # (the created `foo` will set the ENV before exec'ing your executable)
+    env = { PATH: "#{HOMEBREW_PREFIX/"bin"}:$PATH" }
+    (bin/"foo").write_env_script "#{libexec}/bin/foo", env
+
+    # Uncomment if you simply want to symlink the executable – note that this means the first
+    # `node` on the PATH will be used (not necessarily the one Homebrew installed)
+    # bin.install_symlink Dir["#{libexec}/bin/*"]
+
+    # Uncomment if you want to write the completion scripts for bash, fish, and zsh (assuming
+    # your executable has a command "completion" which returns a completion script)
+    # generate_completions_from_executable(libexec/"bin/foo", "completion", base_name: 'foo')
```

Comment:
How many formulae does this?

Reply:
I'm not sure but personally when I was writing a formula I didn't know that this was an option (adding completions) and I also did not know that the `generate_completions_from_executable` function exists. I kept these comments here to help people know that adding completions is easy.

(In my case, my CLI already supported completions – so it was as simple as knowing that it's possible and easy to have Homebrew set it up)

Reply:
Looks like 387 formulae use `generate_completions_from_executable` in total.

However, out of the 173 formulae that have `require "language/node"`, only 10 use `generate_completions_from_executable`.

Of those 10, only [`cdktf`](https://github.com/Homebrew/homebrew-core/blob/01a1f86e06ee66df0b0897613e5b27f553bcf5fb/Formula/c/cdktf.rb#L37-L38) uses the `generate_completions_from_executable(libexec/"bin/foo")` scheme described here, the other 9 use `generate_completions_from_executable(bin/"foo")`.

Reply:
> Of those 10, only [cdktf](https://github.com/Homebrew/homebrew-core/blob/01a1f86e06ee66df0b0897613e5b27f553bcf5fb/Formula/c/cdktf.rb#L37-L38) uses the generate_completions_from_executable(libexec/"bin/foo") scheme described here, the other 9 use generate_completions_from_executable(bin/"foo")

Note that when you use `write_env_script` then you must use `libexec` here (using bin will result in a runtime error)

Reply:
Given this is used by <10% of Node formulae: I'm not sure it's worth documenting specifically here.